test/packetimpact/testbench/connections.go

```diff
@@ -177,6 +177,16 @@ func (conn *TCPIPv4) Send(tcp TCP, additionalLayers ...Layer) {
 	}
 }
 
+// Send a packet with reasonable defaults and override some fields by tcp.
+func (conn *TCPIPv4) SendFrame(Frame Layers, additionalLayers ...Layer) {
+	layersToSend := append(Frame, additionalLayers...)
+	outBytes, err := layersToSend.toBytes()
+	if err != nil {
+		conn.t.Fatalf("can't build outgoing TCP packet: %s", err)
+	}
+	conn.injector.Send(outBytes)
```

Comment:
This will cause problems because it will break the tracking of the sequence numbers that is done in `Send()`.  Can you instead change the prototype of this function to:

`func (conn *TCPIPv4) SendFrame(frame Layers)`

And get rid of the append, let the caller do it.

How I expect that it should work is that `Send` includes lines 154-164 and then SendFrame is the lines 165-177.  And the final line of Send should call SendFrame.  So in essence, the function Send is being broken into two functions, Send and SendFrame, with Send calling SendFrame.

Reply:
[GES] ok Got it. Will take care accordingly